### PR TITLE
initial snapshots stats

### PR DESCRIPTION
**Pull request by sp-joseluis-ledesma**
Hi,

  I took the work sp-francisco-manas was doing, so this PR solves #161

  There are a couple of things to check, and it lacks tests, but I would like to know if you find it misses something, or if there is something you would like to make different before we continue working on this.

regards,

### CODE REVIEW

File: collector/snapshots.go [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+	Value  func(snapshotsStats SnapshotStatsResponse) float64
+	Labels func(repositoryName string) []string
+}
+
+var (
+	defaultSnapshotLabels      = []string{"repository", "state", "version"}
+	defaultSnapshotLabelValues = func(repositoryName string, snapshotStats SnapshotStatDataResponse) []string {
+		return []string{repositoryName, snapshotStats.State, snapshotStats.Version}
+	}
+	defaultSnapshotShardLabels = append(defaultSnapshotLabels, "metric")
```
Comment (zwopir) [inline collector/snapshots.go]: somehow `metric` is a bit to generic, but I'm struggling to find a better name. This is usually not a problem of the one who's trying to find the label name but a problem of the label itself. The label values simply have too less in common. So we should separate the metrics in
- `snapshot_total_shards`
- `snapshot_failed_shards`
- `snapshot_successful_shards`

Another hint for label design is, that it should make sense to sum over the labels. If it doesn't make sense, it shouldn't be in one metric.

Sorry for coming up with this a bit late.

I'd also suggest to drop the `last_` from all the snapshot metrics. Prometheus always scrapes the latest available data, so this seems a bit redundant

**Comment by sp-joseluis-ledesma** [comment]
Hi!

  your comments make sense. I just updated the commit adding also all the missing documentation. I think the only missing part is the tests & travis, right?

regards,

**Comment by mjtrangoni** [comment]
@sp-joseluis-ledesma you can check why Travis is failing locally with `make gometalinter`.

**Comment by sp-joseluis-ledesma** [comment]
If you find OK what I've done I will start with the tests, time to launch a bunch of containers :)

**Comment by zwopir** [comment]
yes, looks good! Some testing would be great

**Comment by sp-joseluis-ledesma** [comment]
I've added tests for the snapshot stats. If there is no further requests from your side, this PR is ready to merge

**Comment by sp-joseluis-ledesma** [comment]
Hi!
  just pushed the change you requested.

**Comment by zwopir** [comment]
thanks a lot for your contribution!

**Pull request merged**